### PR TITLE
impl(generator): full service config parser

### PR DESCRIPTION
**Pull request by coryan**
Parsing via yaml -> jsonproto -> proto works for enums as strings, and
for well-know protos and for a number of other corner cases.

Part of the work for #158

### CODE REVIEW

File: generator/internal/genclient/parser/openapi/openapi.go [hunk 1]
```diff
@@ -151,7 +151,10 @@ func makeServices(api *genclient.API, serviceConfig *serviceconfig.Service, mode
 	if serviceConfig != nil {
 		for _, api := range serviceConfig.Apis {
 			serviceName = api.Name
-			break
+			// Keep searching after well-known mixin services.
+			if !strings.HasPrefix(api.Name, "google.cloud.location.Location") && !strings.HasPrefix(api.Name, "google.iam.v1.IAMPolicy") {
```
Comment (codyoss) [inline generator/internal/genclient/parser/openapi/openapi.go:155]: The other mixin we will need to handle is `google.longrunning.Operations`

File: generator/internal/genclient/service_config.go [hunk 2]
```diff
@@ -18,8 +18,10 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/ghodss/yaml"
 	"google.golang.org/genproto/googleapis/api/serviceconfig"
-	"gopkg.in/yaml.v3"
+	"google.golang.org/protobuf/encoding/protojson"
+	_ "google.golang.org/protobuf/types/known/wrapperspb"
```
Comment (codyoss) [inline generator/internal/genclient/service_config.go:24]: What is this needed for? If it is needed for something could you add a comment.

File: generator/internal/genclient/parser/openapi/openapi.go (hunk 1 above)
Comment (coryan) [reply generator/internal/genclient/parser/openapi/openapi.go]: Fixed.

File: generator/internal/genclient/service_config.go (hunk 2 above)
Comment (coryan) [reply generator/internal/genclient/service_config.go]: Not needed, I neglected to remove this after adding it during troubleshooting. Removed.

**Pull request merged**